### PR TITLE
[5.1] Add resource option to make:controller command

### PR DESCRIPTION
**Pull request by hootlex**
The command

```
php artisan make:controller PostsController --resource=Post
```

Generated controller: PostsController.php

``` php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

use App\Http\Requests;
use App\Http\Controllers\Controller;

class PostsController extends Controller
{
    /**
     * Display a listing of posts
     *
     * @return \Illuminate\Http\Response
     */
    public function index()
    {
        $posts = Post::all();

        return view('posts.index', compact('posts'));
    }

    /**
     * Show the form for creating a new Post
     *
     * @return \Illuminate\Http\Response
     */
    public function create()
    {
        return view('posts.create');
    }

    /**
     * Store a newly created Post in storage.
     *
     * @param  \Illuminate\Http\Request $request
     * @return \Illuminate\Http\Response
     */
    public function store(Request $request)
    {
        Post::create($request->get());

        return Redirect::route('posts.index');
    }

    /**
     * Display the specified Post.
     *
     * @param  int $id
     * @return \Illuminate\Http\Response
     */
    public function show($id)
    {
        $post = Post::findOrFail($id);

        return view('posts.show', compact('post'));
    }

    /**
     * Show the form for editing the specified Post.
     *
     * @param  int  $id
     * @return \Illuminate\Http\Response
     */
    public function edit($id)
    {
        $post = Post::find($id);

        return view('posts.edit', compact('post'));
    }

    /**
     * Update the specified Post in storage.
     *
     * @param  \Illuminate\Http\Request $request
     * @param  int  $id
     * @return \Illuminate\Http\Response
     */
    public function update(Request $request, $id)
    {
        $post = Post::findOrFail($id);

        $post->update($request->get());

        return Redirect::route('posts.index');
    }

    /**
     * Remove the specified Post from storage.
     *
     * @param int $id
     * @return \Illuminate\Http\Response
     */
    public function destroy($id)
    {
        Post::destroy($id);

        return Redirect::route('posts.index');
    }

}
```

### CODE REVIEW

File: src/Illuminate/Routing/Console/stubs/controller.resource.stub [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+class DummyClass extends Controller
+{
+    /**
+     * Display a listing of DummyCollection
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $DummyCollection = DummyModel::all();
```
Comment (phroggyy) [inline src/Illuminate/Routing/Console/stubs/controller.resource.stub]: This won't work. `DummyModel` is referring to the `DummyNamespace`, which is gonna be a controllers namespace. Include a `use DummyModel;` in the top.

Comment (hootlex) [inline src/Illuminate/Routing/Console/stubs/controller.resource.stub]: I will update it in a while.

File: src/Illuminate/Routing/Console/stubs/controller.resource.stub [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        DummyModel::destroy($id);
+
+		return Redirect::route('DummyCollection.index');
+	}
+
+}
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Console/stubs/controller.resource.stub]: minning newline

File: src/Illuminate/Routing/Console/stubs/controller.resource.stub [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $DummyModelVar = DummyModel::findOrFail($id);
+
+		$DummyModelVar->update($request->get());
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Console/stubs/controller.resource.stub]: indentation all over the place

**Comment by taylorotwell** [comment]
Holding off on this.

**Pull request closed without merging**